src/plugins/visualizations/server/plugin.ts

```diff
@@ -37,11 +38,16 @@ export class VisualizationsPlugin
 
   public setup(
     core: CoreSetup,
-    plugins: { usageCollection?: UsageCollectionSetup; embeddable: EmbeddableSetup }
+    plugins: {
+      usageCollection?: UsageCollectionSetup;
+      embeddable: EmbeddableSetup;
+      data: DataPluginSetup;
+    }
   ) {
     this.logger.debug('visualizations: Setup');
 
-    core.savedObjects.registerType(visualizationSavedObjectType);
+    const searchSourceMigrations = plugins.data.search.searchSource.getAllMigrations();
```

Comment:
don't do this, as at this point in time this function is not ready. pass the method in as the dependency rather than the output of the method. `migrations` property of a saved object can be a function which core will call AFTER the setup phase, and only in there its safe to execute `getAllMigrations` method from any other plugin.

Reply:
Thanks for the explanation. In that case, the way I did it in [Lens](https://github.com/elastic/kibana/blob/main/x-pack/plugins/lens/server/plugin.tsx#L54) will need to be updated as well.

Reply:
I have updated these locations to delay loading the `getAllMigrations` functions until `savedObjectType.migrate` is called. However, @flash1293 and I noticed that this isn't currently possible in the context of the embeddable factories. I created an issue here: https://github.com/elastic/kibana/issues/123309